packages/oauth/src/createPublicOauthClient.ts

```diff
@@ -161,8 +161,10 @@ export function createPublicOauthClient(
 
   // as an arrow function, `useHistory` is known to be a boolean
   const go = async (x: string) => {
-    if (useHistory) return window.history.replaceState({}, "", x);
-    else window.location.assign(x);
+    if (useHistory) {
+      window.history.replaceState({}, "", x);
+      return;
```

Comment:
Auto fixer did this when I saved a different change. I kinda like the `return void window...` syntax better but there are two different promise rules and that causes an issue with another one since `.assign` isn't returning a promise, it returns void and so the `void window.history...` is "redundant". But its not cause we need it for the async function. Sigh. So i left this auto fixer as is.